ods_ci/tests/Tests/0200__rhoai_upgrade/0203__post_upgrade.robot

```diff
@@ -59,10 +59,23 @@ Verify RHODS User Groups
     [Documentation]    Verify User Configuration after the upgrade
     [Tags]      Upgrade     Platform        RHOAIENG-19806
     Get Auth Cr Config Data
-    ${admin}                Set Variable            ${AUTH_PAYLOAD[0]['spec']['adminGroups']}
-    ${user}                 Set Variable            ${AUTH_PAYLOAD[0]['spec']['allowedGroups']}
-    Should Be Equal As Strings      '${admin}'      '['${ADMIN_GROUPS}']'
-    Should Be Equal As Strings      '${user}'       '['${ALLOWED_GROUPS}']'
+    @{admins}                Set Variable            ${AUTH_PAYLOAD[0]['spec']['adminGroups']}
+    @{allowed}                 Set Variable            ${AUTH_PAYLOAD[0]['spec']['allowedGroups']}
+
+    ${return_code}    ${adm_groups}=    Run And Return Rc And Output
+    ...    oc get configmap ${UPGRADE_CONFIG_MAP} -n ${UPGRADE_NS} -o jsonpath='{.data.adm_groups}'
+    Should Be Equal As Integers     ${return_code}      0
+
+    ${return_code}    ${allwd_groups}=    Run And Return Rc And Output
+    ...    oc get configmap ${UPGRADE_CONFIG_MAP} -n ${UPGRADE_NS} -o jsonpath='{.data.allwd_groups}'
+    Should Be Equal As Integers     ${return_code}      0
+
+    FOR    ${group}    IN    @{admins}
```

Comment:
Should we be checking `adm_groups` and `allwd_groups` instead?

Reply:
Damn @MarianMacik was quicker :D

Reply:
no, as commented on @MarianMacik thread

Reply:
I don't see this block of code running in execution /4510.
It also looks a little strange, you are iterating over the list of @{admins} but then comparing one of the elements to a scalar of the list. But if @{admins} is empty, which I think it is, this code would never run and give you a false positive.

Reply:
changed to String comparison